jdaviz/configs/cubeviz/plugins/slice/slice.py

```diff
@@ -43,46 +49,38 @@ def __init__(self, *args, **kwargs):
         self.session.hub.subscribe(self, AddDataMessage,
                                    handler=self._on_data_added)
 
-    @observe("linked")
-    def _on_linked_changed(self, event):
-        for viewer in self._watched_viewers:
-
-            if not event['new']:
-                viewer.state.remove_callback('slices',
-                                             self._viewer_slices_changed)
-            else:
+    def _watch_viewer(self, viewer, watch=True):
+        if isinstance(viewer, BqplotImageView):
+            if watch and viewer not in self._watched_viewers:
+                self._watched_viewers.append(viewer)
                 viewer.state.add_callback('slices',
                                           self._viewer_slices_changed)
+            elif not watch and viewer in self._watched_viewers:
+                viewer.state.remove_callback('slices',
+                                             self._viewer_slices_changed)
+                self._watched_viewers.remove(viewer)
+        elif isinstance(viewer, BqplotProfileView) and watch:
+            if self._x_all is None and len(viewer.data()):
+                # cache wavelengths so that wavelength <> slice conversion can be done efficiently
+                self._update_data(viewer.data()[0].spectral_axis)
+
+            if viewer not in self._indicator_viewers:
+                self._indicator_viewers.append(viewer)
+                # if the units (or data) change, we need to update internally
+                viewer.state.add_callback("reference_data",
+                                          self._update_reference_data)
```

Comment:
this logic was moved out of `_on_data_added` and into its own method so it can be called both within `_on_data_added` when processing methods, but also called when initializing the plugin.  This "fixes" slice support to work when calling `Slice(app)` manually (within a test or notebook) after data has already been added.